R/intervals_support_funs.R

```diff
@@ -0,0 +1,363 @@
+
+  # Order the intervals by the index column and then remove it
+  data <- data[order(data[[index_colname]]), ]
+  rownames(data) <- seq_len(nrow(data))
+  data[, !names(data) %in% index_colname]
+}
+
+#' Remove specified imputation from the intervals in a PKNCAdata or data.frame (intervals) object.
+#'
+#' @inheritParams interval_add_impute
+#' @param target_impute A character string specifying the imputation method to remove.
+#' @param ... arguments passed to `interval_remove_impute`.
+#' @return A modified object with the specified imputations removed from the targeted intervals.
+#' @examples
+#' d_conc <- data.frame(
+#'   conc = c(1, 0.6, 0.2, 0.1, 0.9, 0.4, 1.2, 0.8, 0.3, 0.2, 1.1, 0.5),
+#'   time = rep(0:5, 2),
+#'   ID = rep(1:2, each = 6),
+#'   analyte = rep(c("Analyte1", "Analyte2"), each = 6)
+#' )
+#'
+#' d_dose <- data.frame(
+#'   dose = c(100, 200),
+#'   time = c(0, 0),
+#'   ID = c(1, 2)
+#' )
+#'
+#' library(PKNCA)
+#' o_conc <- PKNCA::PKNCAconc(d_conc, conc ~ time | ID / analyte)
+#' o_dose <- PKNCA::PKNCAdose(d_dose, dose ~ time | ID)
+#'
+#' intervals <- data.frame(
+#'   start = c(0, 0, 0),
+#'   end = c(3, 5, Inf),
+#'   half.life = c(TRUE, FALSE, TRUE),
+#'   cmax = c(TRUE, TRUE, TRUE),
+#'   impute = c("start_conc0,start_predose", "start_predose", "start_conc0"),
+#'   analyte = c("Analyte1", "Analyte2", "Analyte1")
+#' )
+#'
+#' o_data <- PKNCAdata(o_conc, o_dose, intervals = intervals)
+#'
+#' # Apply interval_remove_impute function
+#' o_data <- interval_remove_impute(data = o_data,
+#'                                  target_impute = "start_conc0",
+#'                                  target_params = "half.life",
+#'                                  target_groups = data.frame(analyte = "Analyte1"))
+#' @export
+interval_remove_impute <- function(data, ...) {
+  UseMethod("interval_remove_impute", data)
+}
+
+#' @export
+interval_remove_impute.PKNCAdata <- function(data, target_impute, target_params = NULL,
+                                             target_groups = NULL, ...) {
+  # If the impute column is not present in the intervals...
+  if (!"impute" %in% colnames(data$intervals)) {
+    if (is.null(data$impute) || is.na(data$impute)) {
+      # a. If neither is in the global impute method, return the data as it is
+      warning("No default impute column or global method identified. No impute methods to remove")
+      return(data)
+    }
+
+    # b & c. If global impute exists
+    if (isTRUE(!is.na(data$impute))) {
+      if (is.null(target_params) && is.null(target_groups)) {
+        # b. If it is in the global impute and no target parameters or groups, remove global impute
+        data$impute <- remove_impute_method(data$impute, target_impute)
+        return(data)
+      } else {
+        # c. If it is in the global impute but target parameters or groups, creates an impute column
+        data$intervals$impute <- data$impute
+        data$impute <- NA_character_
+      }
+    }
```

Comment:
suggestion: We have some nice early returns, I think we can scrap some of the `if else` statements.
```suggestion
  # If the impute column is not present in the intervals...
  if (!"impute" %in% colnames(data$intervals)) {
    if (is.null(data$impute) || is.na(data$impute)) {
      # a. If neither is in the global impute method, return the data as it is
      warning("No default impute column or global method identified. No impute methods to remove")
      return(data)
    }

    # b & c. If global impute exists
    if (is.null(target_params) && is.null(target_groups)) {
      # b. If it is in the global impute and no target parameters or groups, remove global impute
      data$impute <- remove_impute_method(data$impute, target_impute)
      return(data)
    }

    # c. If it is in the global impute but target parameters or groups, creates an impute column
    data$intervals$impute <- data$impute
    data$impute <- NA_character_
  }
```